frontends/precinct-scanner/src/components/replace_ballot_bag_modal.tsx

```diff
@@ -0,0 +1,110 @@
+import React, { useEffect, useState } from 'react';
+import { Button, Text, Modal } from '@votingworks/ui';
+import { throwIllegalValue } from '@votingworks/utils';
+import { ScannedBallotCount } from './scanned_ballot_count';
+import { CenteredLargeProse } from './layout';
+import { BALLOT_BAG_CAPACITY } from '../config/globals';
+
+interface Props {
+  scannedBallotCount: number;
+  pollWorkerAuthenticated: boolean;
+  onComplete: VoidFunction;
+}
+
+type ModalState = 'no_auth' | 'confirming' | 'confirmed';
+
+export function ReplaceBallotBagModal({
+  scannedBallotCount,
+  pollWorkerAuthenticated,
+  onComplete,
+}: Props): JSX.Element {
+  const [modalState, setModalState] = useState<ModalState>('no_auth');
+
+  useEffect(() => {
+    if (modalState === 'no_auth' && pollWorkerAuthenticated) {
+      setModalState('confirming');
+    } else if (modalState === 'confirming' && !pollWorkerAuthenticated) {
+      setModalState('no_auth');
+    } else if (modalState === 'confirmed' && !pollWorkerAuthenticated) {
+      onComplete();
+    }
+  }, [modalState, onComplete, pollWorkerAuthenticated]);
+
+  const mainContent = (() => {
+    switch (modalState) {
+      case 'no_auth':
+        return (
+          <CenteredLargeProse>
+            <h1>The Ballot Bag is Full</h1>
+            <p>
+              A Poll Worker must replace the full ballot bag with an empty
+              ballot bag.
+            </p>
+            <p>Insert a Poll Worker Card after the ballot bag is replaced.</p>
```

Comment:
A suggestion for making these instructions more straightforward:

> Ask a poll worker to replace the ballot bag. Insert a poll worker card to continue voting.

Reply:
I like the simplified copy, but it doesn't communicate that the poll worker should replace the bag before inserting their card to resume voting.